src/lib/pipeline/m2/material/index.js

```diff
@@ -257,6 +270,58 @@ class M2Material extends THREE.ShaderMaterial {
     }
   }
 
+  registerAnimations(def) {
+    const { uvAnimations, transparencyAnimations, vertexColorAnimation } = def;
+
+    this.registerUVAnimations(uvAnimations);
+    this.registerTransparencyAnimations(transparencyAnimations);
+    this.registerVertexColorAnimation(vertexColorAnimation);
+  }
+
+  registerUVAnimations(uvAnimationIndices) {
+    const { animations, uvAnimationValues } = this.m2;
+
+    uvAnimationIndices.forEach((uvAnimationIndex, opIndex) => {
+      const target = this.uniforms.animatedUVs;
+      const source = uvAnimationValues[uvAnimationIndex];
+
+      animations.on('update', () => {
+        target.value[opIndex] = source.matrix;
+      });
```

Comment:
Sorry, that was badly worded. I was wondering whether all indices could share the same `update`-hook (register a single listener, rather than `uvAnimationIndices.length`.

Yes, using [EventEmitter#removeListener](https://nodejs.org/api/events.html#events_emitter_removelistener_event_listener), 't has some examples :+1: